### DISCUSSION TITLE
Raspberry 4 and USV

### DISCUSSION THREAD

**Discussion by mkollermann** [OpenCCU/OpenCCU#1121] [🇺🇸 Q&A] [answered]
Dear all,

I currently own a CCU3 with a lot of 

- Homematic devices
- Homematic IP devices
- Homematic wired devices
- Homematic wired IP devices.

What I would like to achieve is to have also the use of the on board USV module.

Could you please tell me what (which modules, etc.) I needed to get this running on a Raspberry 4?
Does Cloudmatic still function then also with the raspberrymatic and can I still use the same account (because prepaid)?
Can I just restore the config from CCU3 to the Raspberrymatic?

Thanks a lot for your support in advance !!

BR
Markus

**Comment by jens-maus** [comment, marked as answer]
> What I would like to achieve is to have also the use of the on board USV module.
> 
> Could you please tell me what (which modules, etc.) I needed to get this running on a Raspberry 4?

RaspberryMatic supports a whole bunch of directly HAT attachable USV/UPS models (e.g. S.USV, PiUSV+, etc.) - see https://github.com/jens-maus/RaspberryMatic/wiki/Einleitung#features. Nevertheless, it is probably a better idea to use an external USV/UPS device rather than stuffing a USV /UPS as a HAT attachable module in addition to the GPIO rf module on top of the GPIO. These HAT attachable module could impose RF problems which external UPS devices would of course not cause.

> Does Cloudmatic still function then also with the raspberrymatic and can I still use the same account (because prepaid)?

Yes, if you have a CloudMatic account and use it already with your CCU3 you can perfectly switch and can activate your account also using RaspberryMatic.

> Can I just restore the config from CCU3 to the Raspberrymatic?

Indeed, that's the idea! All you have to make sure is that you switch from the same 3.X.Y firmware version from a CCU3 to RapberryMatic. So if you are, e.g. currently using 3.55.5 with your CCU3 you can perfectly switch to a 3.55.5.YYYYMMDD version of RaspberryMatic. But if you are on 3.55.10 already with your CCU3 you will currently have to wait a few days until a 3.55.10.YYYYMMDD RaspberryMatic has been released.

**Reply by mkollermann** [reply]
Vielen Dank für die rasche Antwort !!👏🙏😀
